nautobot/core/utils/filtering.py

```diff
@@ -114,9 +114,13 @@ def get_filterset_parameter_form_field(model, parameter, filterset=None):
     elif isinstance(field, NumberFilter):
         form_field = forms.IntegerField()
     elif isinstance(field, ModelMultipleChoiceFilter):
+        if getattr(field, "prefers_pk", False):
```

Comment:
This is a tough one because it's just moving the problem around. In 1.x we had separate filters for `vlan__vid` and `vlan__id` and we had the ability to filter on either one in the UI. This is "fixing" the problem by removing the ability to filter on `vlan__vid` from the UI and making it so you can only filter on `vlan__id`. Perhaps if we encounter a NaturalKeyOrPKMultipleChoiceFilter we need to render two form fields?

Reply:
Is there a defined criteria for choosing the filters that should have `prefers_pk` or is it on a case by case basis? If it's not consistent should we just make the behavior change global so all `NaturalKeyOrPKMultipleChoiceFilter`s use the PK?

Reply:
My thought was any filter that doesn't have a single-field natural key (globally unique `name` or similar) should have `prefers_pk`. Those that have a single-field natural key can reasonably use that field name.